app/src/main/java/com/infomaniak/swisstransfer/ui/screen/main/received/ReceivedScreen.kt

```diff
@@ -157,7 +163,37 @@ private fun ReceivedScreen(areTransfersEmpty: () -> Boolean) {
             TransferItemList(
                 modifier = Modifier.padding(Margin.Medium),
                 transfers = transfers,
-                onClick = { /* TODO */ }
+                onClick = { transfer ->
+                    when {
+                        transfer.expiresInDays < 0 -> {
+                            isVisible = true
+                            expirationDate = Date(transfer.expirationDateTimestamp)
+                        }
+                        transfer.downloadLeft == 0 -> {
+                            isVisible = true
+                            downloadsLimit = transfer.downloadLimit
+                        }
+                        else -> {
+                            Log.d("TODO", "Display transfer details screen")
+                            // TODO
+                        }
+                    }
+                }
+            )
+
+            TransferExpiredBottomSheet(
```

Comment:
Maybe, having a `if (isExpired)` instead of `isVisible` for the condition might be clearer. That way, we know the `BottomSheet` is only visible if the transfer is expired.